### ISSUE TITLE
[Future -  To Do] Set Up CI/CD for Linting and Prettier Rules Validation on Pull Requests

### ISSUE THREAD

**Issue by spiritanand** [llmshare/llmproto#11]
## Issue Description

**Problem:**
Currently, our codebase may not adhere to consistent linting and formatting standards. This can lead to inconsistent code quality and style, making it challenging for developers to maintain and review the code. Inconsistent code styling can also introduce subtle bugs and reduce code readability.

**Solution:**
To address this issue and ensure that all code contributions meet our linting and formatting standards, we propose setting up a Continuous Integration/Continuous Deployment (CI/CD) pipeline that automatically validates linting and Prettier rules on each pull request (PR) before it is merged into the main codebase.

## Expected Behaviour

When a developer opens a pull request, the CI/CD pipeline should be triggered automatically. This pipeline will perform the following tasks:

1. **Linting Rules Validation:** The CI/CD pipeline should run our preferred linting tool (e.g., ESLint) to analyse the code changes in the PR. It should identify any linting errors or warnings and report them back to the PR.

2. **Prettier Rules Validation:** The pipeline should also run Prettier on the code changes to ensure consistent code formatting. If there are any formatting issues, Prettier should be configured to automatically format the code.

3. **Status Reporting:** The CI/CD pipeline should provide clear status indications on the PR, indicating whether the code adheres to linting and formatting rules and whether the build and tests have passed or failed.

## Acceptance Criteria

- [ ] The CI/CD pipeline is configured to trigger automatically upon opening a new pull request.
- [ ] The pipeline runs ESLint to check for linting errors and warnings.
- [ ] Prettier is used to enforce code formatting rules, and any formatting issues are automatically fixed.
- [ ] The PR status reflects the success or failure of linting, formatting, build, and tests.
- [ ] Detailed reports of linting and formatting issues are available in the PR comments.
- [ ] Developers receive feedback in a timely manner to address linting and formatting issues.

## Definition of Done

- [ ] CI/CD pipeline is configured and integrated with the repository.
- [ ] The pipeline checks for linting and formatting errors on PRs.
- [ ] Linting and formatting issues are reported in PR comments.
- [ ] The code automatically adheres to linting and Prettier rules.
- [ ] The PR status accurately reflects the results of linting and formatting.
- [ ] Developers receive clear feedback on the status of their PRs.
- [ ] The CI/CD setup is documented for future reference.

## Note

This issue can serve as a central point for discussion and tracking progress toward implementing the requested CI/CD improvements. Feel free to add any additional context, suggestions, or requirements.

**Comment by Sing-Li** [comment]
@spiritanand  please *do not* start on this. Right now our project is far from mature,  adding this will immediately add out-of-box complexity AND stop new contributions dead. let's get our project into a better shape before we start on this.
